src/managers/GuildInviteManager.js

```diff
@@ -0,0 +1,185 @@
+   */
+
+  add(data, cache) {
+    return super.add(data, cache, { id: data.code, extras: [this.guild] });
+  }
+
+  /**
+   * Data that resolves to give a Invite object. This can be:
+   * * An invite code
+   * * An invite URL
+   * @typedef {string} InviteResolvable
+   */
+
+  /**
+   * Resolves a InviteResolvable to a Invite object.
+   * @method resolve
+   * @memberof GuildInviteManager
+   * @instance
+   * @param {InviteResolvable} invite The invite resolvable to resolve
+   * @returns {?Invite}
+   */
+
+  /**
+   * Resolves a InviteResolvable to a invite code string.
+   * @method resolveID
+   * @memberof GuildInviteManager
+   * @instance
+   * @param {InviteResolvable} invite The invite resolvable to resolve
+   * @returns {?string}
+   */
+
+  /**
+   * Options used to fetch a single invite from a guild.
+   * @typedef {Object} FetchInviteOptions
+   * @property {InviteResolvable} code The invite to fetch
+   * @property {boolean} [cache=true] Whether or not to cache the fetched invite
+   * @property {boolean} [force=false] Whether to skip the cache check and request the API
+   */
+
+  /**
+   * Options used to fetch all invites from a guild.
+   * @typedef {Object} FetchInvitesOptions
+   * @property {boolean} cache Whether or not to cache the fetched invites
+   */
+
+  /**
+   * Fetches invite(s) from Discord.
+   * @param {InviteResolvable|FetchInviteOptions|FetchInvitesOptions} [options] Options for fetching guild invite(s)
+   * @returns {Promise<Invite>|Promise<Collection<Snowflake, Invite>>}
+   * @example
+   * // Fetch all invites from a guild
+   * guild.invites.fetch()
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch all invites from a guild without caching
+   * guild.invites.fetch({ cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite
+   * guild.invites.fetch('bRCvFy9')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite without checking cache
+   * guild.invites.fetch({ code: 'bRCvFy9', force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single invite without caching
+   * guild.invites.fetch({ code: 'bRCvFy9', cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
+    if (!options) return this._fetchMany();
+    if (typeof options === 'string') {
+      const code = DataResolver.resolveInviteCode(options);
+      if (!code) return Promise.reject(new Error('INVITE_RESOLVE_CODE'));
+      return this._fetchSingle({ code, cache: true });
+    }
+    if (!options.code) {
+      if ('cache' in options) return this._fetchMany(options.cache);
+      return Promise.reject(new Error('INVITE_RESOLVE_CODE'));
+    }
+    return this._fetchSingle({
+      ...options,
+      code: DataResolver.resolveInviteCode(options.code),
+    });
+  }
```

Comment:
I was thinking about this fetch method options 🤔 
We don't have any channel parameter, if we add one to options, we can call this method from `GuildChannel#fetchInvites` and cache received invites from `GET /channels/{channel.id}/invites`.

Reply:
`{ channelID: Snowflake } | { code: Snowflake }` sounds good to me, perhaps remove the string overload to prevent ambiguity.

Just make sure to define those two overloads separately since they call different methods and return different data... and that they can't be used together in the types.

Reply:
I can take care of it but I would need help with typing.
I did not understand how to separate them into two "overloads"

Reply:
Something like the following:
```typescript
public fetch(options: { ..., channelID: Snowflake }): Promise<Collection<Snowflake, Invite>>;
public fetch(options: string | { ..., code: string }): Promise<Invite>;
```

Reply:
Is it not just possible to edit `FetchInvitesOptions` with optional channelID?
The returned value is the same 🤔 
```ts
interface FetchInvitesOptions {
  channelID?: Snowflake
  cache: boolean;
}
```

Reply:
Sure thing, as long as the second overload takes strictly either a string (invite code), or an object with a required code.

Also, is cache not optional?

Reply:
> Also, is cache not optional?
Yep, I change it 👍